src/node/index.d.ts

```diff
@@ -0,0 +1,526 @@
+   *   ProtoBuf.js is provided in the value argument. If the option is 'detect',
+   *   gRPC will guess what the version is based on the structure of the value.
+   *   Defaults to 'detect'.
+   * @param {Object} value The ProtoBuf.js reflection object to load
+   * @param {Object=} options Options to apply to the loaded file
+   * @return {Object<string, *>} The resulting gRPC object
+   */
+  export function loadObject(replectionObject: ReflectionObject, options?: ILoadProtobufOptions): IProtobufDefinition;
+
+  /**
+   * Serer object that stores request handlers and delegates incoming requests to those handlers
+   */
+  export class Server {
+    /**
+     * Constructs a server object
+     * @param options Options that should be passed to the internal server implementation
+     */
+    constructor(options?: IServerOptions);
+
+    /**
+     * Start the server and begin handling requests
+     */
+    start(): void;
+
+    /**
+     * Gracefully shuts down the server. The server will stop receiving new calls,
+     * and any pending calls will complete. The callback will be called when all
+     * pending calls have completed and the server is fully shut down. This method
+     * is idempotent with itself and forceShutdown.
+     * @param callback The shutdown complete callback
+     */
+    tryShutdown(callback: () => void): void;
+
+    /**
+     * Forcibly shuts down the server. The server will stop receiving new calls
+     * and cancel all pending calls. When it returns, the server has shut down.
+     * This method is idempotent with itself and tryShutdown, and it will trigger
+     * any outstanding tryShutdown callbacks.
+     */
+    forceShutdown(): void;
+
+    /**
+     * Registers a handler to handle the named method. Fails if there already is
+     * a handler for the given method. Returns true on success
+     * @param name The name of the method that the provided function should handle/respond to.
+     * @param handler Function that takes a stream of request values and returns a stream of response values
+     * @param serialize Serialization function for responses
+     * @param deserialize Deserialization function for requests
+     * @param type The streaming type of method that this handles
+     * @return True if the handler was set. False if a handler was already set for that name.
+     */
+    register(name: string, handler: () => void, serialize: (obj: any) => Buffer, deserialize: (buffer: Buffer) => any, type: string): boolean;
+
+    /**
+     * Add a service to the server, with a corresponding implementation. If you are
+     * generating this from a proto file, you should instead use `addProtoService`.
+     * @param service The service descriptor, as `getProtobufServiceAttrs` returns
+     * @param implementation Map of method names to method implementation for the provided service.
+     */
+    addService(service: { [index: string]: any }, implementation: { [index: string]: () => void }): void;
```

Comment:
You might be right. The arguments are specified by the given service method arguments, right?